construct/core.py

```diff
@@ -1549,18 +1552,31 @@ def __init__(self, keyfunc, cases, default=NoDefault, includekey=False):
         self.cases = cases
         self.default = default
         self.includekey = includekey
-        self.flagbuildnone = all(sc.flagbuildnone for sc in cases.values())
-        self.flagembedded = all(sc.flagembedded for sc in cases.values())
+        all_cases = list(cases.values())
+        if default is not self.NoDefault:
+            all_cases.append(default)
+        self.flagbuildnone = all(sc.flagbuildnone for sc in all_cases if sc is not Pass)
+        self.flagembedded = all(sc.flagembedded for sc in all_cases if sc is not Pass)
```

Comment:
Should it be embedded iff embedded for all subconstructs?
I think `Switch` should choose one of the subcons and be completely transparent otherwise.
As it stands, this implementation brings a bit of subtle unexpected behaviour of its own, that depends on the *other* subcons:

```python
example = Struct(
    'choice' / Byte,
    'inner' / Switch(
        lambda ctx: ctx.choice,
        {
            1: Struct('not_embedded' / Int16ub),
            5: Embedded(Struct('embedded' / Int16ub))
        }
    ))
# This works as expected
example.parse(b"\x01\x01\x02")
# The Embedded Struct gets its flag ignored.
example.parse(b"\x05\x01\x02")
# But if all subcons are Embedded then it looks like this:
Container(choice=5)(embedded=258)
```